src/install/install.zig

```diff
@@ -4305,11 +4310,32 @@ pub const PackageManager = struct {
                     var k: usize = 0;
 
                     while (k < new_dependencies.len) : (k += 1) {
+                        if (new_dependencies[k].key) |key| {
+                            if (key.data.e_string.eql(string, update.name)) {
+                                if (update.resolved_name.isEmpty()) {
+                                    // This actually is a duplicate
+                                    // like "react" appearing in both "dependencies" and "optionalDependencies"
+                                    // For this case, we'll just swap remove it
+                                    if (new_dependencies.len > 1) {
+                                        new_dependencies[k] = new_dependencies[new_dependencies.len - 1];
+                                        new_dependencies = new_dependencies[0 .. new_dependencies.len - 1];
+                                    } else {
+                                        new_dependencies = &[_]G.Property{};
+                                    }
+                                    continue;
+                                }
+                                new_dependencies[k].key = null;
+                            }
+                        }
+
                         if (new_dependencies[k].key == null) {
                             new_dependencies[k].key = JSAst.Expr.init(
                                 JSAst.E.String,
                                 JSAst.E.String{
-                                    .data = update.name,
+                                    .data = if (update.resolved_name.isEmpty())
+                                        update.name
+                                    else
+                                        try allocator.dupe(u8, update.resolved_name.slice(update.version_buf)),
```

Comment:
This will give me gibberish if I don't `.dupe()` it...

Reply:
E.String is recycled memory